sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeOperationTests.cs

```diff
@@ -0,0 +1,410 @@
+
+            var keyPhrasesListId2 = new List<string> { "Tesla stock", "year" };
+
+            // TODO - Update this to result[0].KeyPhrases
+            foreach (string keyphrase in result[1].KeyPhrases)
+            {
+                Assert.IsTrue(keyPhrasesListId2.Contains(keyphrase));
+            }
+        }
+
+        [Test]
+        public async Task AnalyzeOperationWithTopParameter()
+        {
+            TextAnalyticsClient client = GetClient();
+
+            AnalyzeOperationOptions operationOptions = new AnalyzeOperationOptions()
+            {
+                KeyPhrasesTaskParameters = new KeyPhrasesTaskParameters()
+                {
+                    ModelVersion = "latest"
+                },
+                DisplayName = "AnalyzeOperationWithSkipParameter",
+                Top = 1
+            };
+
+            AnalyzeOperation operation = await client.StartAnalyzeOperationBatchAsync(batchConvenienceDocuments, "en", operationOptions);
+
+            await operation.WaitForCompletionAsync();
+
+            AnalyzeOperationResult resultCollection = operation.Value;
+
+            ExtractKeyPhrasesResultCollection result = resultCollection.KeyPhraseResult;
+
+            Assert.IsNotNull(result);
+
+            // TODO - Update this to Assert.AreEqual(1, result.Count), once skip starts working.
+            Assert.AreEqual(2, result.Count);
+
+            var keyPhrasesListId1 = new List<string> { "CEO of SpaceX", "Elon Musk", "Tesla" };
+
+            foreach (string keyphrase in result[0].KeyPhrases)
+            {
+                Assert.IsTrue(keyPhrasesListId1.Contains(keyphrase));
+            }
+        }
+
+        [Test]
+        public async Task AnalyzeOperationBatchWithErrorTest()
+        {
+            TextAnalyticsClient client = GetClient();
+
+            var documents = new List<string>
+            {
+                "Subject is taking 100mg of ibuprofen twice daily",
+                "Can cause rapid or irregular heartbeat, delirium, panic, psychosis, and heart failure.",
+                "",
+            };
+
+            AnalyzeOperationOptions operationOptions = new AnalyzeOperationOptions()
+            {
+                KeyPhrasesTaskParameters = new KeyPhrasesTaskParameters()
+                {
+                    ModelVersion = "latest"
+                },
+                DisplayName = "AnalyzeOperationBatchWithErrorTest",
+                Skip = 1
+            };
+
+            var exceptionMessage = "At least one document is missing a Text attribute.\r\nStatus: 400 (Bad Request)\r\nErrorCode: InvalidArgument\r";
+            await Task.Run(() => {
+                RequestFailedException ex = Assert.ThrowsAsync<RequestFailedException>(async () =>
+                {
+                   AnalyzeOperation operation = await client.StartAnalyzeOperationBatchAsync(documents, "en", operationOptions);
+                });
+
+                Assert.IsTrue(ex.Message.Contains(exceptionMessage));
```

Comment:
could you check the error code instead of the error message?
Error message is a string, which means service can change it whenever. If they do, they will break the test.
The service can't change ErrorCode without it being a breaking change, so we can rely on that value